pkg/resource/deployment.go

```diff
@@ -82,6 +83,17 @@ func (gd *generatorDeployment) expected() (runtime.Object, error) {
 	}
 	podTemplateSpec.Annotations[parameters.ChecksumOperatorDepsAnnotation] = depsChecksum
 
+	strategy := gd.cr.Spec.RolloutStrategy
+	// We support only two possible rollout strategies: RollingUpdate and Recreate
+	if strategy != "RollingUpdate" && strategy != "Recreate" {
+		return nil, fmt.Errorf("Unsupported rollout strategy type: %v", strategy)
+	}
+
+	// if we use Recreate rollout strategy, then we must have only one image registry replica
+	if gd.cr.Spec.RolloutStrategy == "Recreate" && gd.cr.Spec.Replicas != 1 {
+		return nil, fmt.Errorf("Not possible to use more than 1 replica with Recreate rollout strategy")
```

Comment:
isn't this going to start failing existing clusters?  tightening validation rules is very problematic.

also why can't you use recreate with 1 replica?  (yes you'll have an outage, but it's technically possible/allowed).

Reply:
Agree, we need one replica when we want to upgrade, not here.
I'll update the patch in a minute.